### ISSUE TITLE
Severe and Urgent Error

### ISSUE THREAD

**Issue by SoldatulBanana** [SpigotMC/BungeeCord#1854]
00:36:07 [SEVERE] [/92.83.189.231:56111] <-> InitialHandler - encountered exception
io.netty.handler.codec.DecoderException: java.lang.IndexOutOfBoundsException: readerIndex(35) + length(1) exceeds writerIndex(35): SlicedAbstractByteBuf(ridx: 35, widx: 35, cap: 35/35, unwrapped: PooledUnsafeDirectByteBuf(ridx: 36, widx: 57, cap: 512))
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:99)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:318)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:304)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:276)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:354)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:244)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:318)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:304)
	at io.netty.handler.timeout.ReadTimeoutHandler.channelRead(ReadTimeoutHandler.java:152)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:318)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:304)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:846)
	at io.netty.channel.epoll.AbstractEpollStreamChannel$EpollStreamUnsafe.epollInReady(AbstractEpollStreamChannel.java:823)
	at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:339)
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:255)
	at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:112)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.IndexOutOfBoundsException: readerIndex(35) + length(1) exceeds writerIndex(35): SlicedAbstractByteBuf(ridx: 35, widx: 35, cap: 35/35, unwrapped: PooledUnsafeDirectByteBuf(ridx: 36, widx: 57, cap: 512))
	at io.netty.buffer.AbstractByteBuf.checkReadableBytes0(AbstractByteBuf.java:1178)
	at io.netty.buffer.AbstractByteBuf.readByte(AbstractByteBuf.java:587)
	at net.md_5.bungee.protocol.DefinedPacket.readVarInt(DefinedPacket.java:89)
	at net.md_5.bungee.protocol.DefinedPacket.readVarInt(DefinedPacket.java:79)
	at net.md_5.bungee.protocol.DefinedPacket.readArray(DefinedPacket.java:50)
	at net.md_5.bungee.protocol.packet.EncryptionResponse.read(EncryptionResponse.java:26)
	at net.md_5.bungee.protocol.MinecraftDecoder.decode(MinecraftDecoder.java:33)
	at net.md_5.bungee.protocol.MinecraftDecoder.decode(MinecraftDecoder.java:10)
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:89)
	... 16 more
00:36:07 [WARNING] [/92.83.189.231:56118] <-> InitialHandler - IOException: syscall:read(...)() failed: Connection reset by peer

Basically this will full the proxy.log and the server will stop writing then crash.

What do you think?
Tested on about 5 networks with 400+ users, all down

**Comment by iKaneki-Ken** [comment]
Someone is attacking your server.

**Comment by i9hdkill** [comment]
Fixed in newer builds
